assets/src/blocks/Submenu/SubmenuEditor.js

```diff
@@ -69,11 +71,64 @@ const renderEdit = (attributes, setAttributes) => {
   );
 }
 
-const renderView = (attributes, setAttributes) => {
+// We can put the other blocks that can have a header inside in here along with the attribute containing the heading text.
+// Then we can also filter those to include them in the menu.
+const blockTypesWithHeadings = [
+  {name: 'planet4-blocks/articles', fieldName: 'article_heading', level: 2},
+];
 
-  const { menuItems } = useSubmenuItemsLoad(attributes.levels, true);
 
-  const style = getSubmenuStyle(attributes.className, attributes.submenu_style);
+const renderView = (attributes, setAttributes, className) => {
+
+  const { blocks } = useSelect(select => {
+    return ({ blocks: select('core/editor').getBlocks() });
+  });
+
+  const { levels: selectedLevels } = attributes
+
+  const flatHeaders = [];
+  blocks.forEach(block => {
```

Comment:
I still need to move some things from here to a separate function to keep `SubmenuEditor` a bit more high level.

Reply:
done :heavy_check_mark: